core/array.cpp

```diff
@@ -355,6 +355,48 @@ Variant Array::pop_front() {
 	return Variant();
 }
 
+Array *Array::reduce(Object *p_obj, const StringName &p_function, const Variant &p_args) {
+	Array *ret = new Array();
```

Comment:
There is no reason to use heap allocation for the Array object here, you can use `Array ret;` and it will work just fine

Reply:
ok, I change it.
I did this because travis complaint about it, with out heap allocation: [Travis](https://travis-ci.org/godotengine/godot/jobs/409497012)
But I may read it wrong. I am not realy expierienced with C++.

Reply:
You can check how other GDNative functions deal with it, you usually write to a separate local variable, then create a GDNative-type variable and copy the raw data over.